overlord/patch/patch6_3.go

```diff
@@ -0,0 +1,99 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package patch
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+func normChan(in string) string {
+	return strings.Join(strings.FieldsFunc(in, func(r rune) bool { return r == '/' }), "/")
```

Comment:
Two comments: please describe in a comment how channel normalisation works and why we need to perform it. It's much better to understand the *intent* than to derive one from the implementation, especially as that latter one drifts.

Second: is this just `filepath.Clean`? I know it's probably not 100% exactly that but it has a well-known semantics and might be a cleaner way to express the intent.